pandas/tests/io/json/test_json_table_schema.py

```diff
@@ -435,6 +435,23 @@ def test_to_json_categorical_index(self):
 
         assert result == expected
 
+    @pytest.mark.parametrize("name", [None, "foo"])
+    def test_multiindex_single_level(self, name):
+        # GH29928
+        index = pd.Index([1, 2, 3, 4], name=name)
+        df = pd.DataFrame(
+            data=[[1, 1], [2, 2], [3, 3], [4, 4]], columns=["A", "B"], index=index
+        )
+        expected = df.to_json(orient="table")
+
+        index = pd.MultiIndex.from_tuples([(1,), (2,), (3,), (4,)], names=[name])
+        df = pd.DataFrame(
+            data=[[1, 1], [2, 2], [3, 3], [4, 4]], columns=["A", "B"], index=index
+        )
+        result = df.to_json(orient="table")
+
+        assert result == expected
```

Comment:
do these fully round-trip? can you test that

Reply:
They should, I will complete tests.